src/Sylius/Behat/Page/Shop/Product/ShowPage.php

```diff
@@ -145,121 +95,91 @@ public function getAttributeByName(string $name): ?string
         return trim($row->find('css', 'td.sylius-product-attribute-value')->getText());
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getAttributes()
+    public function getAttributes(): array
     {
         $attributesTable = $this->getElement('attributes');
 
         return $attributesTable->findAll('css', 'tr > td.sylius-product-attribute-name');
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function hasProductOutOfStockValidationMessage(ProductInterface $product)
+    public function getAverageRating(): float
     {
-        $message = sprintf('%s does not have sufficient stock.', $product->getName());
-
-        if (!$this->hasElement('validation_errors')) {
-            return false;
-        }
+        return (float) $this->getElement('average_rating')->getAttribute('data-average-rating');
+    }
 
-        return $this->getElement('validation_errors')->getText() === $message;
+    public function getCurrentUrl(): string
+    {
+        return $this->getDriver()->getCurrentUrl();
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function waitForValidationErrors($timeout)
+    public function getCurrentVariantName(): string
     {
-        $errorsContainer = $this->getElement('selecting_variants');
+        $currentVariantRow = $this->getElement('current_variant_input')->getParent()->getParent();
 
-        $this->getDocument()->waitFor($timeout, function () use ($errorsContainer) {
-            return false !== $errorsContainer->has('css', '[class ~="sylius-validation-error"]');
-        });
+        return $currentVariantRow->find('css', 'td:first-child')->getText();
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getPrice()
+    public function getName(): string
     {
-        return $this->getElement('product_price')->getText();
+        return $this->getElement('name')->getText();
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function countReviews()
+    public function getPrice(): string
     {
-        return count($this->getElement('reviews')->findAll('css', '.comment'));
+        return $this->getElement('product_price')->getText();
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function hasReviewTitled($title)
+    public function hasAddToCartButton(): bool
     {
-        return null !== $this->getElement('reviews')->find('css', sprintf('.comment:contains("%s")', $title));
+        return $this->getDocument()->hasButton('Add to cart')
```

Comment:
Something like this? 
```suggestion
        return 
            $this->getDocument()->hasButton('Add to cart') &&
            false === $this->getDocument()->findButton('Add to cart')->hasAttribute('disabled')
        ;
```
I'm wondering if we cannot fetch a button, check if it is null and then check if it has disabled attribute in separate checks, as it will be easier to read